source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -107,6 +107,7 @@ void EnvoyQuicClientStream::encodeTrailers(const Http::RequestTrailerMap& traile
 void EnvoyQuicClientStream::encodeMetadata(const Http::MetadataMapVector& /*metadata_map_vector*/) {
   // Metadata Frame is not supported in QUIC.
   // TODO(danzh): add stats for metadata not supported error.
+  ENVOY_BUG(false, "METADATA is not supported in Http3.");
```

Comment:
I think HTTP/1.1 just swallows it with a stat increase.  I'd mildly prefer that to crashing in debug mode.